### DISCUSSION TITLE
Truly Fully Automated Switching Mode

### DISCUSSION THREAD

**Discussion by ZhuGene** [seerge/g-helper#2800] [Ideas]
Many thanks to the author seerge for his selfless contribution, to develop such a useful g-helper, which allows us, the free riders, can uninstall the extremely difficult-to-use Armoury Crate. I didn't quite understand the community rules before, so I first posted a question in Discussions and then, upon seerge's reminder, created an Issue. The point is that I realized the keyboard lighting issue might have been caused by my own oversight in not fully disabling ASUS services 🙈. 

I have a new request that I dare bring up to seerge here: 

My current settings are to use Turbo Mode when plugged in for performance and Silent Mode when unplugged, with the graphics mode set to automatically switch between Standard Mode (iGPU + dGPU) when plugged in and iGPU Mode when unplugged. However sometimes, when I need to use an external HDMI monitor or projector during a meeting while unplugged, I have to manually switch the graphics mode to Standard Mode because for ROG external display devices are directly connected to the dedicated GPU. Is it possible to automatically switch to Standard Mode when I plug in an HDMI cable, essentially enhancing the auto-switching mode to become a truly fully automated switching mode? 🐶

**Comment by seerge** [comment]
@ZhuGene hello, this is a duplicate of https://github.com/seerge/g-helper/issues/909 . I was answering this before - there is no reliable way to detect when HDMI is plugged.

If you use external monitor - either use left USB-C port driven by iGPU or use `Standard` mode all the time.

**Reply by ZhuGene** [reply]
OK, thank u.
